x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/components/access_control_index_selector/access_control_index_selector.tsx

```diff
@@ -7,76 +7,110 @@
 
 import React from 'react';
 
-import { EuiFlexGrid, EuiFlexItem, EuiSuperSelect, EuiText, EuiTitle } from '@elastic/eui';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiSuperSelect,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 
 export interface AccessControlSelectorOption {
   description: string;
+  error: boolean;
   title: string;
   value: 'content-index' | 'access-control-index';
 }
 
-const indexSelectorOptions: AccessControlSelectorOption[] = [
-  {
-    description: i18n.translate(
-      'xpack.enterpriseSearch.content.searchIndex.documents.selector.contentIndex.description',
-      {
-        defaultMessage: 'Browse content fields',
-      }
-    ),
-    title: i18n.translate(
-      'xpack.enterpriseSearch.content.searchIndex.documents.selector.contentIndex.title',
-      {
-        defaultMessage: 'Content index',
-      }
-    ),
-    value: 'content-index',
-  },
-  {
-    description: i18n.translate(
-      'xpack.enterpriseSearch.content.searchIndex.documents.selector.accessControl.description',
-      {
-        defaultMessage: 'Browse document level security fields',
-      }
-    ),
-    title: i18n.translate(
-      'xpack.enterpriseSearch.content.searchIndex.documents.selector.accessControl.title',
-      {
-        defaultMessage: 'Access control index',
-      }
-    ),
-    value: 'access-control-index',
-  },
-];
-
 interface IndexSelectorProps {
+  accessControlIndexDescription?: string;
+  accessControlIndexTitle?: string;
+  accessSyncError?: boolean;
+  contentIndexDescription?: string;
+  contentIndexTitle?: string;
+  contentSyncError?: boolean;
+  fullWidth?: boolean;
   onChange(value: AccessControlSelectorOption['value']): void;
   valueOfSelected?: AccessControlSelectorOption['value'];
 }
 
 export const AccessControlIndexSelector: React.FC<IndexSelectorProps> = ({
-  valueOfSelected,
+  accessControlIndexDescription,
+  accessControlIndexTitle,
+  accessSyncError,
+  contentIndexDescription,
+  contentIndexTitle,
+  contentSyncError,
   onChange,
+  valueOfSelected,
+  fullWidth,
 }) => {
+  const indexSelectorOptions: AccessControlSelectorOption[] = [
+    {
+      description: i18n.translate(
+        'xpack.enterpriseSearch.content.searchIndex.documents.selector.contentIndex.description',
+        {
+          defaultMessage: contentIndexDescription || 'Browse content fields',
```

Comment:
Done and working, nice architecture! Just curiosity, one thing I don't quite see is that now when we pass a custom `indexSelectorOptions` how it can get and apply the passed prop? if we do:
`indexSelectorOptions = DEFAULT_INDEX_SELECTOR_OPTIONS,`
Which I guess it should always set the value to the default one again. Trying to understand better this pattern on appling default values when no props are passed.

Reply:
in the context of passing arguments to the functions (Or react components which are functions) if you set a parameter that is optional. The syntax can be confusing but here is a simpler example. 

```ts

interface MyFuncArgs {
    arg1: number,
    arg2?: number
}

// arg2 is optional.
// arg2 = 5 means when arg2 is not passed, use 5 as arg2
// without it it would be undefined
function myFunc({arg1, arg2 = 5}: MyFuncArgs) {
 console.log("arg1: ", arg1);
 console.log("arg2: ", arg2);
}


myFunc({arg1: 3}); 
// arg1: 3 
// arg2: 5

myFunc({ arg1: 3, arg2: 3}); 
// arg1: 3 
// arg2: 3


```

For our case, if we don't pass `indexSelectorOptions` at all, we will use `DEFAULT_INDEX_SELECTOR_OPTIONS`. 
If user wants to add/remove/change/partially change something, they have to pass their complete options. For partial, we export the defaults where they can copy and change it and pass it back.